blockmanager.go

```diff
@@ -873,6 +878,7 @@ func (b *blockManager) getCheckpointedCFHeaders(checkpoints []*chainhash.Hash,
 	//
 	// TODO(roasbeef): extract to func to test
 	currentInterval := startingInterval
+	endHeightRange := curHeight
```

Comment:
Why's this change necessary?

Reply:
Earlier the `queryBatch` method would call the response handle every time a new response came in. Now this code is not part of the response handler anymore, and we must know when to exit the loop.